### PR TITLE
Enhancement/pdf web

### PR DESCRIPTION
**Pull request by Yunnigg**

### CODE REVIEW

File: pronto_mia/lib/app/app.router.dart [hunk 1]
```diff
@@ -113,20 +112,17 @@ class DeploymentPlanOverviewViewArguments {
...
 }
 
 /// PdfView arguments holder class
 class PdfViewArguments {
   final Key key;
   final String title;
   final String subTitle;
-  final FileUpload pdfUpload;
-  final String pdfPath;
+  final Object pdfFile;
```
[the remark this answers is not in the document]
Comment (Yunnigg) [reply pronto_mia/lib/app/app.router.dart:125]: I have merged pdfUpload and pdfPath to pdfFile to be able to enforce it as required. This leads to it being either of type String or FileUpload. The closest ancestor of both of these types is Object.

File: pronto_mia/lib/ui/views/pdf/pdf_viewmodel.dart [hunk 2]
```diff
@@ -1,42 +1,43 @@
-import 'dart:io';
+import 'package:flutter/cupertino.dart';
```
[the remark this answers is not in the document]
Comment (Yunnigg) [reply pronto_mia/lib/ui/views/pdf/pdf_viewmodel.dart]: It is not needed, I replaced it with `foundation.dart`.

There are a few things that are defined in `foundation.dart` (core flutter primitives), `cupertino.dart` and `material.dart` at the same time, such as the `@required` annotation for named parameters. Android Studio's auto import prefers `cupertino.dart` for some reason, which in this case is not problematic but also not entirely correct either.

File: pronto_mia/lib/app/app.router.dart (hunk 1 above)
[the remark this answers is not in the document]
Comment (Vec7or) [reply pronto_mia/lib/app/app.router.dart:125]: Ok, not ideal but understandable.

File: pronto_mia/lib/ui/views/pdf/pdf_view.dart [hunk 3]
```diff
@@ -1,32 +1,26 @@
...
-import 'package:pronto_mia/core/models/file_upload.dart';
 import 'package:pronto_mia/ui/views/pdf/pdf_viewmodel.dart';
+import 'package:pronto_mia/ui/components/data_view_layout.dart';
 
 class PdfView extends StatelessWidget {
   final String title;
   final String subTitle;
-  final FileUpload pdfUpload;
-  final String pdfPath;
+  final Object pdfFile;
```
[the remark this answers is not in the document]
Comment (Vec7or) [reply pronto_mia/lib/ui/views/pdf/pdf_view.dart:11]: Ok, not ideal but understandable.

**Pull request merged**